homeassistant/components/isy994/services.py

```diff
@@ -326,40 +324,31 @@ async def async_set_variable_service_handler(service: ServiceCall) -> None:
     def async_cleanup_registry_entries(service: ServiceCall) -> None:
         """Remove extra entities that are no longer part of the integration."""
         entity_registry = er.async_get(hass)
-        config_ids = []
-        current_unique_ids: set[str] = set()
 
         for config_entry_id in hass.data[DOMAIN]:
             entries_for_this_config = er.async_entries_for_config_entry(
                 entity_registry, config_entry_id
             )
-            config_ids.extend(
-                [
-                    (entity.unique_id, entity.entity_id)
-                    for entity in entries_for_this_config
-                ]
+            entities = {
+                (entity.domain, entity.unique_id): entity.entity_id
```

Comment:
Im still a bit confused by the evolution of `domain` and `platform`, but it seems like this is misnamed in `RegistryEntry`. `entity.domain` returns the platform (e.g. `sensor`)

Reply:
Its reversed for config entry/top-level ones because everything used to be set up from yaml and was something like

`sensor.isy994`

```
sensor: 
  - platform: isy994
```

 but now everything is `isy994.sensor`

```
isy994:
```

etc